src/core/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -665,14 +598,49 @@ export class SavedObjectsClient {
    * }
    * ```
    */
-  createPointInTimeFinder(
+  createPointInTimeFinder<T = unknown, A = unknown>(
     findOptions: SavedObjectsCreatePointInTimeFinderOptions,
     dependencies?: SavedObjectsCreatePointInTimeFinderDependencies
-  ): ISavedObjectsPointInTimeFinder {
+  ): ISavedObjectsPointInTimeFinder<T, A> {
     return this._repository.createPointInTimeFinder(findOptions, {
       client: this,
       // Include dependencies last so that SO client wrappers have their settings applied.
       ...dependencies,
     });
   }
+
+  /**
+   * Gets all references and transitive references of the listed objects. Ignores any object that is not a multi-namespace type.
+   *
+   * @param objects
+   * @param options
+   */
+  async collectMultiNamespaceReferences(
+    objects: SavedObjectsCollectMultiNamespaceReferencesObject[],
+    options?: SavedObjectsCollectMultiNamespaceReferencesOptions
+  ): Promise<SavedObjectsCollectMultiNamespaceReferencesResponse> {
```

Comment:
The PR is not really introducing it, but we have an inconsistency on the `space` vs `namespace` naming. eg `collectMultiNamespaceReferences` versus `updateObjectsSpaces`

Reply:
Yeah, it was my understanding that we want to eventually get rid of `namespace` terminology in favor of `space` now that we aren't shipping an OSS distribution. I agree this is confusing, especially because this is the first mention of "Spaces" in the Core codebase. I'm happy to change this to `updateObjectsNamespaces` if you think that would be better for now.